server/sonar-server/src/test/java/org/sonar/server/projectlink/ws/SearchActionTest.java

```diff
@@ -0,0 +1,202 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.core.permission.GlobalPermissions.SYSTEM_ADMIN;
+import static org.sonar.core.util.Uuids.UUID_EXAMPLE_01;
+import static org.sonar.server.ws.KeyExamples.KEY_PROJECT_EXAMPLE_001;
+import static org.sonar.test.JsonAssert.assertJson;
+import static org.sonarqube.ws.MediaTypes.PROTOBUF;
+import static org.sonarqube.ws.client.projectlinks.ProjectLinksWsParameters.PARAM_PROJECT_ID;
+import static org.sonarqube.ws.client.projectlinks.ProjectLinksWsParameters.PARAM_PROJECT_KEY;
+
+public class SearchActionTest {
+
+  private final String PROJECT_KEY = KEY_PROJECT_EXAMPLE_001;
+  private final String PROJECT_UUID = UUID_EXAMPLE_01;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Rule
+  public UserSessionRule userSession = UserSessionRule.standalone();
+
+  @Rule
+  public DbTester db = DbTester.create(System2.INSTANCE);
+  DbClient dbClient = db.getDbClient();
+  DbSession dbSession = db.getSession();
+
+  WsActionTester ws;
+
+  SearchAction underTest;
+
+  @Before
+  public void setUp() {
+    ComponentFinder componentFinder = new ComponentFinder(dbClient);
+    underTest = new SearchAction(dbClient, userSession, componentFinder);
+    ws = new WsActionTester(underTest);
+
+    userSession.login("login").setGlobalPermissions(SYSTEM_ADMIN);
+  }
+
+  @Test
+  public void test_example() {
+    ComponentDto project =  insertProject();
+    insertHomepageLink(project.uuid());
+    insertCustomLink(project.uuid());
+
+    String result = ws.newRequest()
+      .setParam(PARAM_PROJECT_KEY, PROJECT_KEY)
+      .execute().getInput();
+
+    assertJson(result).ignoreFields("id").isSimilarTo(getClass().getResource("list-example.json"));
+  }
+
+  @Test
+  public void request_by_project_id() throws IOException {
+    ComponentDto project =  insertProject();
+    insertHomepageLink(project.uuid());
+
+    InputStream responseStream = ws.newRequest()
+      .setParam(PARAM_PROJECT_ID, PROJECT_UUID)
+      .setMediaType(PROTOBUF)
+      .execute().getInputStream();
+
+    WsProjectLinks.SearchWsResponse response = WsProjectLinks.SearchWsResponse.parseFrom(responseStream);
+
+    assertThat(response.getLinksCount()).isEqualTo(1);
+    assertThat(response.getLinks(0).getName()).isEqualTo("Homepage");
+  }
+
+  @Test
+  public void request_by_project_key() throws IOException {
+    ComponentDto project =  insertProject();
+    insertHomepageLink(project.uuid());
+
+    InputStream responseStream = ws.newRequest()
+      .setParam(PARAM_PROJECT_KEY, PROJECT_KEY)
+      .setMediaType(PROTOBUF)
+      .execute().getInputStream();
+
+    WsProjectLinks.SearchWsResponse response = WsProjectLinks.SearchWsResponse.parseFrom(responseStream);
+
+    assertThat(response.getLinksCount()).isEqualTo(1);
+    assertThat(response.getLinks(0).getName()).isEqualTo("Homepage");
+  }
+
```

Comment:
add a test with links from different projects, to check it returns only links from the specified project